C builder error when trying to make a channel with non-defined type

I use latest V version on windows 10.

When trying to make a channel with non-existent type (sctruct or builtin type)
v produces C builder errror
```v
fn main() {
	_ := chan NonExistingType{}
}
```
this is enough to test this
(Also cause of this #14752)

error:
```
C:/Users/Sasha/AppData/Local/Temp/v_0/channel-test.13407381772732692168.tmp.c:3294: error: ';' expected (got "__chan_main__NonExistingType_popval")
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -3515,6 +3515,12 @@ pub fn (mut c Checker) chan_init(mut node ast.ChanInit) ast.Type {
 	if node.typ != 0 {
 		info := c.table.sym(node.typ).chan_info()
 		node.elem_type = info.elem_type
+		if node.elem_type != 0 {
+			elem_sym := c.table.sym(node.elem_type)
+			if elem_sym.kind == .placeholder {
+				c.error('unknown type `$elem_sym.name`', node.pos)
+			}
+		}
 		if node.has_cap {
 			c.check_array_init_para_type('cap', node.cap_expr, node.pos)
 		}

--- a/vlib/v/checker/tests/chan_elem_type_unknown.out
+++ b/vlib/v/checker/tests/chan_elem_type_unknown.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/chan_elem_type_unknown.vv:2:7: error: unknown type `NonExistingType`
+    1 | fn main() {
+    2 |     _ := chan NonExistingType{}
+      |          ~~~~~~~~~~~~~~~~~~~~~~
+    3 | }

--- a/vlib/v/checker/tests/chan_elem_type_unknown.vv
+++ b/vlib/v/checker/tests/chan_elem_type_unknown.vv
@@ -1,0 +1,3 @@
+fn main() {
+	_ := chan NonExistingType{}
+}

